src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,268 @@
+            'field': 'Name of the field to retrieve, required',
+            'count': 'Whether to retrieve the count of matching events',
+            'timestamp_field': 'Name of the timestamp field.'
+                               'Defaults to \'timestamp\'',
+            'timestamp_format': 'Timestamp format as for strptime or \'int\''
+                                'for unix time. Defaults to \'int\'',
+            'subdomain': 'Loggly subdomain to retrieve results from',
+            'query': 'Loggly search query',
+            'username': 'Loggly username',
+            'password': 'Loggly password'
+        })
+        return config_help
+
+    def process_config(self):
+        """Prepare metrics' configuration."""
+        def set_defaults(config):
+            defaults = {
+                'timestamp_field': 'timestamp',
+                'timestamp_format': 'int',
+                'field': 'count',
+                'count': False,
+                'query': '*'
+            }
+            [config.update({key: value}) for key, value in defaults.items()
+                if key not in config.keys()]
+
+        super(LogglyCollector, self).process_config()
+
+        self.config_bunch = bunch.bunchify(self.config)
+        self.metrics_config = self.config_bunch.metric
+
+        for metric, config in self.metrics_config.items():
+            if 'field' not in config.keys():
+                raise DiamondException(
+                    'field must be specified for the metric %s in file %s' %
+                    (metric, self.configfile))
+
+            set_defaults(config)
+
+            config.update({
+                'search_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/search?q={0.query}&from='
+                              '-{1.interval}s&until=now'.format(
+                                  config, self.config_bunch),
+                'events_URI': 'https://{0.subdomain}.loggly.com'
+                              '/apiv2/events?rsid=%s'.format(config)})
+
+    def collect(self):
+        """Collect metrics from loggly."""
+        def parse_timestamp(metric_config, timestamp):
+            if metric_config.timestamp_format is 'int':
+                return datetime.datetime.fromtimestamp(
+                    timestamp / 10**(ceil(log10(timestamp)) - 10))
+            else:
+                return datetime.datetime.strptime(timestamp,
+                                                  self.config_bunch
+                                                  .timestamp_format)
+
+        def retrieve_results(metric_config):
+            def now_as_timefield():
+                if metric_config.timestamp_format is 'int':
+                    return time.mktime(datetime.datetime.now().timetuple())
+                else:
+                    return datetime.datetime.now().strftime(
+                        metric_config.timestamp_format)
+
+            def log_error_and_continue(code):
```

Comment:
Loggly has its own explanation for their error codes, most match the standard reason, but not necessarily, these come from their API documentation.